### PR TITLE
Added element-wise multiplication as merge mode

### PR DESCRIPTION
**Pull request by jfsantos**
In a graph model, this would allow having a branch generating a mask to another branch, which can be useful in many models.

### CODE REVIEW

**Comment by iskandr** [comment]
:+1: This would be useful for a model I'm working on. 

This does raise the question of whether the set of merges should be more flexibly extensible. Another merge I'd like to have (related to this) is the matrix product (e.g. one branch of the network outputs a 2D matrix and the other a vector to multiple that array). Should I add that as "matmul" or should the merge mode also accept function arguments? 

Also, why is elementwise product limited to two layers? You could conceivably filter by multiple masks simultaneously.

**Comment by jfsantos** [comment]
@iskandr thanks for the comments! There's no reason for the elementwise product to be limited to two layers. I did it just to be sure I would have only one "masker" and one "masked" layer, but indeed elementwise multiplication could be performed in any number of layers.

I think it would be nice to have the merge mode to accept function arguments, as it would make adding arbitrary merge modes much easier.

File: keras/layers/core.py [hunk 1]
```diff
@@ -163,10 +163,12 @@ def get_config(self):
 class Merge(Layer):
     def __init__(self, layers, mode='sum'):
         ''' Merge the output of a list of layers or containers into a single tensor.
-            mode: {'sum', 'concat'}
+            mode: {'sum', 'mul', 'concat'}
         '''
         if len(layers) < 2:
             raise Exception("Please specify two or more input layers (or containers) to merge")
+        elif len(layers) > 2 and mode == 'mul':
+            raise Exception("Elemwise multiplication mode can only merge two layers")
```
Comment (fchollet) [inline keras/layers/core.py]: This should definitely be changed to support N inputs (at no loss of functionality...), much like `sum`.

Comment (jfsantos) [inline keras/layers/core.py]: Fixed in last commit, thanks!

Comment (fchollet) [inline keras/layers/core.py]: Awesome : ) But now you'll need to remove the outdated exception as well ; )

**Pull request merged**